lib/lte_link_control/lte_lc.c

```diff
@@ -334,18 +337,60 @@ static void at_handler_xt3412(const char *response)
 	event_handler_list_dispatch(&evt);
 }
 
-static void at_handler_ncellmeas(const char *response)
+static void at_handler_ncellmeas_gci(const char *response)
 {
 	int err;
 	struct lte_lc_evt evt = {0};
+	const char *resp = response;
 
 	__ASSERT_NO_MSG(response != NULL);
 
-	int ncell_count = neighborcell_count_get(response);
-	struct lte_lc_ncell *neighbor_cells = NULL;
+	int max_cell_count = ncellmeas_params.gci_count;
+	struct lte_lc_cell *cells = NULL;
 
-	LOG_DBG("%%NCELLMEAS notification");
-	LOG_DBG("Neighbor cell count: %d", ncell_count);
+	LOG_DBG("%%NCELLMEAS GCI notification parsing starts");
+	if (max_cell_count != 0) {
+		cells = k_calloc(max_cell_count, sizeof(struct lte_lc_cell));
+		if (cells == NULL) {
+			LOG_ERR("Failed to allocate memory for the GCI cells");
+			return;
+		}
+	}
+
+	evt.cells_info.gci_cells = cells;
+	err = parse_ncellmeas_gci(&ncellmeas_params, resp, &evt.cells_info);
+
+	switch (err) {
+	case -E2BIG:
+		LOG_WRN("Not all neighbor cells could be parsed");
+		LOG_WRN("More cells than the configured max count of %d were found",
+			CONFIG_LTE_NEIGHBOR_CELLS_MAX);
+		/* Fall through */
+	case 0: /* Fall through */
+	case 1:
+		LOG_DBG("Neighbor cell count: %d, GCI cells count: %d",
+			evt.cells_info.ncells_count,
+			evt.cells_info.gci_cells_count);
+			evt.type = LTE_LC_EVT_NEIGHBOR_CELL_MEAS;
```

Comment:
Fix indentation or is it fine in editor?

Reply:
Fixed.